Mathlib/SetTheory/Ordinal/Arithmetic.lean

```diff
@@ -1058,109 +1058,204 @@ theorem comp_familyOfBFamily {o} (f : ∀ a < o, α) (g : α → β) :
-  sup_le fun i =>
-    match h (mem_range_self i) with
-    | ⟨_j, hj⟩ => hj ▸ le_sup _ _
+  Ordinal.sSup_le_sSup h (bddAbove_range.{v, max u w} _)
 
-theorem sup_eq_of_range_eq {ι ι'} {f : ι → Ordinal} {g : ι' → Ordinal}
+-- TODO: generalize or remove
+theorem iSup_eq_of_range_eq {ι ι'} {f : ι → Ordinal} {g : ι' → Ordinal}
+    (h : Set.range f = Set.range g) : iSup f = iSup g :=
+  congr_arg _ h
+
+set_option linter.deprecated false in
+@[deprecated iSup_eq_of_range_eq (since := "2024-08-27")]
+theorem sup_eq_of_range_eq {ι : Type u} {ι' : Type v}
+    {f : ι → Ordinal.{max u v w}} {g : ι' → Ordinal.{max u v w}}
     (h : Set.range f = Set.range g) : sup.{u, max v w} f = sup.{v, max u w} g :=
-  (sup_le_of_range_subset.{u, v, w} h.le).antisymm (sup_le_of_range_subset.{v, u, w} h.ge)
+  Ordinal.iSup_eq_of_range_eq h
 
-@[simp]
-theorem sup_sum {α : Type u} {β : Type v} (f : α ⊕ β → Ordinal) :
-    sup.{max u v, w} f =
-      max (sup.{u, max v w} fun a => f (Sum.inl a)) (sup.{v, max u w} fun b => f (Sum.inr b)) := by
-  apply (sup_le_iff.2 _).antisymm (max_le_iff.2 ⟨_, _⟩)
+-- TODO: generalize to conditionally complete lattices
+theorem iSup_sum {α : Type u} {β : Type v} (f : α ⊕ β → Ordinal.{max u v w}) :
+    iSup f = max (⨆ a, f (Sum.inl a)) (⨆ b, f (Sum.inr b)) := by
+  apply (Ordinal.iSup_le _).antisymm (max_le _ _)
   · rintro (i | i)
-    · exact le_max_of_le_left (le_sup _ i)
-    · exact le_max_of_le_right (le_sup _ i)
+    · exact le_max_of_le_left (Ordinal.le_iSup.{u, max u v w} _ i)
+    · exact le_max_of_le_right (Ordinal.le_iSup.{v, max u v w} _ i)
   all_goals
-    apply sup_le_of_range_subset.{_, max u v, w}
+    apply Ordinal.sSup_le_sSup _ (bddAbove_range _)
     rintro i ⟨a, rfl⟩
     apply mem_range_self
 
+set_option linter.deprecated false in
+@[deprecated iSup_sum (since := "2024-08-27")]
+theorem sup_sum {α : Type u} {β : Type v} (f : α ⊕ β → Ordinal.{max u v w}) :
+    sup.{max u v, w} f =
+      max (sup.{u, max v w} fun a => f (Sum.inl a)) (sup.{v, max u w} fun b => f (Sum.inr b)) :=
+  iSup_sum f
+
+theorem unbounded_range_of_iSup_ge {α β : Type u} (r : α → α → Prop) [IsWellOrder α r] (f : β → α)
+    (h : type r ≤ ⨆ i, typein r (f i)) : Unbounded r (range f) :=
+  (not_bounded_iff _).1 fun ⟨x, hx⟩ =>
+    h.not_lt <| lt_of_le_of_lt
+      (Ordinal.iSup_le fun y => ((typein_lt_typein r).2 <| hx _ <| mem_range_self y).le)
+      (typein_lt_type r x)
+
+set_option linter.deprecated false in
+@[deprecated unbounded_range_of_iSup_ge (since := "2024-08-27")]
 theorem unbounded_range_of_sup_ge {α β : Type u} (r : α → α → Prop) [IsWellOrder α r] (f : β → α)
     (h : type r ≤ sup.{u, u} (typein r ∘ f)) : Unbounded r (range f) :=
-  (not_bounded_iff _).1 fun ⟨x, hx⟩ =>
-    not_lt_of_le h <|
-      lt_of_le_of_lt
-        (sup_le fun y => le_of_lt <| (typein_lt_typein r).2 <| hx _ <| mem_range_self y)
-        (typein_lt_type r x)
+  unbounded_range_of_iSup_ge r f h
```

Comment:
Just for your understanding, this part of the diff is good: The statement of `unbounded_range_of_sup_ge` is unchanged.